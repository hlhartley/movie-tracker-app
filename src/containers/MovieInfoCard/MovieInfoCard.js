import React from 'react';

export const MovieInfoCard = ({ title, id, release_date, poster_path, overview }) => {
  return(
    <div className='movie-info'>
      <div>
        <img src={`https://image.tmdb.org/t/p/w500/${poster_path}`} className='movie-info-poster'/>
      </div>
      <div>
        <h2>{title}</h2>
        <p className='released-text'>Released: {release_date}</p>
        <p>{overview}</p>
      </div>
    </div>
  )
}