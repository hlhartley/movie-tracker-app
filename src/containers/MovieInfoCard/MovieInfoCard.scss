.movie-info {
    display: flex;
    background-color: $brown;
    border: 3px solid transparent;
    border-radius: 5px;
    width: 80%;
    text-align: center;
    padding: 20px;
    margin: 25px;
    p {
        margin-top: 10px;
    }
    h2 {
        color: $gold;
    }
}

.movie-info-poster {
    width: 300px;
    margin-right: 15px;
}

.released-text {
    font-weight: bold;
}
