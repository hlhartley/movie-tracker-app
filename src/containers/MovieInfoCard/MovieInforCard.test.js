import React from 'react'; 
import { shallow } from 'enzyme';
import { MovieInfoCard } from './MovieInfoCard';

describe('MovieInfoCard', () => {
  it('should match the wrapper with all data passed in', () => {
    const mockMovie = {
      id: 375588,
      title: "Robin Hood",
      poster_path: "/AiRfixFcfTkNbn2A73qVJPlpkUo.jpg",
      overview: "A war-hardened Crusader and his Moorish commander mount an audacious revolt against the corrupt English crown.",
      release_date: "2018-11-20",
    }
    const wrapper = shallow(<MovieInfoCard {...mockMovie}/>);
    expect(wrapper).toMatchSnapshot();
  });
});