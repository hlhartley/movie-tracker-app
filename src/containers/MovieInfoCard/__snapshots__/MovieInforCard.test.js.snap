// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfoCard should match the wrapper with all data passed in 1`] = `
<div
  className="movie-info"
>
  <div>
    <img
      className="movie-info-poster"
      src="https://image.tmdb.org/t/p/w500//AiRfixFcfTkNbn2A73qVJPlpkUo.jpg"
    />
  </div>
  <div>
    <h2>
      Robin Hood
    </h2>
    <p
      className="released-text"
    >
      Released: 
      2018-11-20
    </p>
    <p>
      A war-hardened Crusader and his Moorish commander mount an audacious revolt against the corrupt English crown.
    </p>
  </div>
</div>
`;
